tokio-executor/src/enter.rs

```diff
@@ -10,8 +11,7 @@ thread_local!(static ENTERED: Cell<bool> = Cell::new(false));
 ///
 /// For more details, see [`enter` documentation](fn.enter.html)
 pub struct Enter {
-    on_exit: Vec<Box<dyn Callback>>,
-    permanent: bool,
+    _p: PhantomData<RefCell<()>>,
```

Comment:
Sorry to bother. I'm reading the source code recently and I'm really curious about the usage of `PhantomData` here. After some research, it looks like a trick to make `Enter` `Send` but not `Sync`. If so, why not explicitly declare it as `!Sync`. Is there any historical reason or did I miss some other reasons why `PhantomData<RefCell<()>>` is used here?

Reply:
Probably just habit.

Reply:
Thanks for your reply. I have figured out another cause, negative trait bounds are not yet fully implemented, we can only use marker types to mark a type that is not `Send` or `Sync`.